src/osp/Resource/AssetImporter.cpp

```diff
@@ -67,20 +67,23 @@ void osp::AssetImporter::load_sturdy_file(std::string const& filepath, Package&
         std::cout << "Error: couldn't open file: " << filepath << "\n";
     }
 
-    load_sturdy(gltfImporter, pkg);
+    // repurpose filepath into unique identifier for file resources
+    std::string dataResourceName = filepath.substr(0, filepath.find('.')) + ':';
+
+    load_sturdy(gltfImporter, dataResourceName, pkg);
 
     gltfImporter.close();
 }
 
 void osp::AssetImporter::load_part(TinyGltfImporter& gltfImporter,
-    Package& pkg, unsigned id)
+    Package& pkg, unsigned id, std::string_view resPrefix)
 {
     // It's a part
     std::cout << "PART!\n";
 
     // Recursively add child nodes to part
     PrototypePart part;
-    proto_add_obj_recurse(gltfImporter, pkg, part, 0, id);
+    proto_add_obj_recurse(gltfImporter, pkg, resPrefix, part, 0, id);
 
     // Parse extra properties
     tinygltf::Node const& node = *static_cast<tinygltf::Node const*>(
```

Comment:
i notice nothing actually ever uses "node" other than the line where you bind node.extras to a reference.

You can use structured bindings for that.

tinygltf::Value const& [extras] = *static_cast<...>(...);

Reply:
Can you briefly explain how structured bindings work here? Is `extras` an arbitrary name, or is it accessing the `node.extras` member? The node has a bunch of `Value`s in it, so I don't understand how the compiler would be able to figure out what we're trying to do in this case.

Reply:
Actually, structured bindings may not work here.

Structured bindings work using the machinery for std::get<>, where the first thing in the structured binding is index 0, and the second is index 1, and so on.

Better would be to instead do

```
tinygltf::Value const& extras = static_cast<...>(...)->extras;
```
directly.

Sorry for the confusion.

Reply:
Alright makes sense, thanks for the clarification